code/datums/components/subtype_picker.dm

```diff
@@ -0,0 +1,88 @@
+/**
+ * subtype picker component allows for an item to transform into its subtypes (this is not enforced and you can turn in whatever types, but
+ * i used this name as it was incredibly accurate for current usage of the behavior)
+ *
+ * Used for the null rod to pick the other holy weapons.
+ */
+/datum/component/subtype_picker
+	///on Attach, it's a list of types and their menu descriptions. after building, it's a list of lists with all the data each item needs to give to the radial menu.
```

Comment:
I really dislike this pattern. Wait you're not even modifying `subtype2descriptions` later what does this mean

Reply:
incorrect doc, sorry. just having it not mutate makes this less dislikable, yes?

Reply:
yes very much. vars meaning one thing is more then enough